src/OrdinaryDiffEq.jl

```diff
@@ -37,7 +37,8 @@ module OrdinaryDiffEq
                      resize_non_user_cache!,deleteat_non_user_cache!,addat_non_user_cache!,
                      terminate!,get_du, get_dt,get_proposed_dt,set_proposed_dt!,
                      u_modified!,savevalues!,add_tstop!,add_saveat!,set_reltol!,
-                     set_abstol!, postamble!, last_step_failed
+                     set_abstol!, postamble!, last_step_failed,
+                     isautodifferentiable
```

Comment:
I don't think this needs to be exported?

Reply:
I think it's handled by `@reexport using DiffEqBase`.

```
julia> using OrdinaryDiffEq

help?> isautodifferentiable
search: isautodifferentiable

  No documentation found.

  DiffEqBase.isautodifferentiable is a Function.

  # 2 methods for generic function "isautodifferentiable":
  isautodifferentiable(alg::OrdinaryDiffEq.OrdinaryDiffEqAlgorithm) in OrdinaryDiffEq at /home/takafumi/.julia/v0.6/OrdinaryDiffEq/src/alg_utils.jl:1
  isautodifferentiable(alg::DiffEqBase.DEAlgorithm) in DiffEqBase at /home/takafumi/.julia/v0.6/DiffEqBase/src/alg_traits.jl:1
```

Reply:
Oops. I read your sentence wrongly. But `isautodifferentiable` is jsut imported from `DiffEqBase` so I suppose it's fine?

If you don't want to export `isautodifferentiable` from OrdinaryDiffEq, I guess we need to avoid `@reexport` in OrdinaryDiffEq.jl or remove `export isautodifferentiable` from DiffEqBase.jl.

Reply:
Yeah, exporting less from DifferentialEquations packages sounds like a good idea.  I personally like explicit import (`using DiffEqBase: stuff, I, need`) to make code clearer.  Just an idea, but one way to get rid of `export` gradually is to go this explicit import style. Once it's done for all packages, `export` would become irrelevant so you can make it minimal.

Reply:
Yeah, it's the upgrading the packages that's a lot of work. I still want to export the public API, but all of those abstract types isn't necessary to export.